### ISSUE TITLE
Lint reports 3DS2 and Redirect components being library restricted

### ISSUE THREAD

**Issue by zsoltvilagos** [Adyen/adyen-android#1372]
**Describe the bug**

`./gradlew lintDebug` reports error when code is to get `Adyen3DS2Component` or `RedirectComponent` with their respective `PROVIDER.get(...)`.

**To Reproduce**
1. Go to Adyen SDK example app, e.g. `CardActivity`
2. Navigate to `setupCardView`
3. Paste the following at the start of the method:
    ```
    RedirectComponent.PROVIDER.get<Any>(
        activity = this,
        configuration = RedirectConfiguration.Builder(this, Environment.TEST, clientKey = "fake").build(),
        callback = object : ActionComponentCallback {
            override fun onAdditionalDetails(actionComponentData: ActionComponentData) {
            }
            override fun onError(componentError: ComponentError) {
            }
        },
        key = null,
    )
    ```
4. On repo root, execute `./gradlew :example-app:lintDebug`
5. See failure:
    ```
    /Users/vilagoszs/Work/adyen/repos/adyen-android/example-app/src/main/java/com/adyen/checkout/example/ui/card/CardActivity.kt:98: Error: ActionComponentProvider.get can only be called from within the same library group (referenced groupId=com.adyen.checkout from groupId=adyen-android) [RestrictedApi]
            val redirectComponent = RedirectComponent.PROVIDER.get<Any>(
   ```

**Expected behavior**
`lintDebug` shall succeed because using `PROVIDER.get(...)` should be the way to get the component. 

**Screenshots**
none

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: N/A
 - Version: macOS Sonoma 14.0

**Smartphone (please complete the following information):**
 - Device: N/A
 - OS: N/A
 - Browser: N/A
 - Version: N/A

**Additional context**
With SDK 4.x, it was working. I experienced the issue while migrating to 5.0.0.

I think because `ActionComponentProvider` is annotated with `@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)`, `Adyen3DS2Component` and `RedirectComponent` are as well.

**Comment by zsoltvilagos** [comment]
A related thing: the `ActionComponentProvider.get(...)` overload that takes an `Activity` is generic with type parameter `T` but that type parameter is not used. See https://github.com/Adyen/adyen-android/blob/5.0.0/components-core/src/main/java/com/adyen/checkout/components/core/internal/provider/ActionComponentProvider.kt#L75

Would that be maybe some leftover?

**Comment by jreij** [comment]
Hi @zsoltvilagos, thanks for reporting this and glad to see you're already migrating to v5! 🎉 

You're actually spot on with both issues, the first one should not be restricted and the generic was left there by mistake. We will run some more tests and fix them 👍 For the moment you can simply ignore these warnings and hopefully we'll release a fix soon.

For more context, Android Studio had a bug where it wouldn't show you a warning for using restricted APIs and lint was inconsistent, detecting some things but not all of them. We managed to clean up most annotations that shouldn't have been there but I guess a few of them slipped. Anyway we've learned that the bug is fixed in the upcoming Android Studio Hedgehog so that should make it easier to detect these without lint.

**Comment by jreij** [comment]
And FYI, you can now use the `GenericActionComponent` instead of both `Adyen3DS2Component`  and `RedirectComponent`, if it makes your implementation easier.

**Comment by zsoltvilagos** [comment]
> And FYI, you can now use the `GenericActionComponent` instead of both `Adyen3DS2Component` and `RedirectComponent`, if it makes your implementation easier.

I've set that as a next step. Already investigate its code a lot to see how to set it up.

> 
> Or even simpler, you can use the `CardComponent` alone to handle 3DS2 and the redirect, with the `handleAction` method.

I'm set to to investigate that component 👍

**Comment by jreij** [comment]
Hi @zsoltvilagos, we just released [5.0.1](https://github.com/Adyen/adyen-android/releases/tag/5.0.1) which includes a fix for this issue, let us know if you still face any problems with it.

**Comment by zsoltvilagos** [comment]
> Hi @zsoltvilagos, we just released [5.0.1](https://github.com/Adyen/adyen-android/releases/tag/5.0.1) which includes a fix for this issue, let us know if you still face any problems with it.

Thank you for the heads-up. Already bumped, CI 🟢 .
